thorlcr/graph/thgraph.hpp

```diff
@@ -1171,6 +1172,17 @@ class graph_decl CActivityBase : implements CInterfaceOf<IThorRowInterfaces>, im
     IThorRowInterfaces * createRowInterfaces(IOutputMetaData * meta, byte seq=0);
     IThorRowInterfaces * createRowInterfaces(IOutputMetaData * meta, roxiemem::RoxieHeapFlags heapFlags, byte seq=0);
 
+    CFileSizeTracker * queryTempFileSizeTracker()
+    {
+        if (!tempFileSizeTracker)
+            tempFileSizeTracker.setown(new CFileSizeTracker);
+        return tempFileSizeTracker;
+    }
+    offset_t queryActiveTempSize() const
+    {
+        return tempFileSizeTracker ? tempFileSizeTracker->queryActiveSize() : 0;
```

Comment:
I think this is only updated when the file is closed.  @jakesmith I assume that is going to work for hash dedup, but will it scale to other situations - e.g. lookahead - where the file is not closed until the end.

Also are you planning to publish the active and peak spill file size for each activity?  What should we call them rather than spill files?  Do we need to rename the existing internal spills to match this convention.

Reply:
>What should we call them rather than spill files? 

we are restricting 'spill' to mean inter-subgraph spills. And 'temp' to mean any internal activity temp files usage (i.e. inc. lookahead etc.). NB: we already do distinguish that way via "spill" and "temp" plane categories.

Reply:
>I think this is only updated when the file is closed. 

the tracker could be updated at any time, but yes in hashdedup, it only updates it at the end, when a bucket spills. That seemed reasonable since the write is expected to be relatively quick.

For lookahead, we should update periodically not at end.

The actually collection is periodic (in this PR) and will collect whatever updates there are.

Reply:
Lookahead will be done differently, so that the size is updated intermittently.

I'm not attempting to record active temp file size as at the active temp size always ends up 0 because at the end of each activity the temp files are closed.

Reply:
Active could be useful, e.g. a long running subgraph that hit peak early on, but is now slow because active is growing slowly.
We should also capture the total temp written too. e.g. peak might be a few GB, but it may have written near to that 100's of times.
Can revisit in another PR though.